lambdas/shared/t4_lambda_shared/decorator.py

```diff
@@ -0,0 +1,58 @@
+"""
+Decorators for using lambdas in API Gateway
+"""
+
+from functools import wraps
+import traceback
+
+from jsonschema import Draft4Validator, ValidationError
+
+def api(cors_origins=[]):
+    def innerdec(f):
+        @wraps(f)
+        def wrapper(event, _):
+            params = event['queryStringParameters'] or {}
+            headers = event['headers'] or {}
+            try:
+                status, body, response_headers = f(params, headers)
+            except Exception:
+                traceback.print_exc()
+                status = 500
+                body = 'Internal Server Error'
+                response_headers = {
+                    'Content-Type': 'text/plain'
+                }
+
+            origin = headers.get('origin')
+            if origin is not None and origin in cors_origins:
+                response_headers.update({
+                    'access-control-allow-origin': '*',
+                    'access-control-allow-methods': 'GET,POST',
```

Comment:
might need HEAD at some point... I guess preview in doesn't need HEAD, so maybe @api gets a default kwarg `allow_methods=['GET', 'POST']`. i'm not religious about this but, at least for interacting with S3, we keep needing more and more methods than we expected.